tests/cypress/e2e/helpers.ts

```diff
@@ -114,3 +114,23 @@ export function activateCmkChanges(siteName: string) {
     waitForActivation(activation_id);
   });
 }
+
+export function recursiveType(
+  targetSelector: string,
+  checkSelector: string,
+  bodySelector: string,
+  typedCommand: string,
+  count: number,
+  forceType: boolean
+) {
+  cy.log('Recursive typing ' + typedCommand + ' in ' + targetSelector + ' (' + count + ' tries left)');
+  cy.get(targetSelector).type(typedCommand, { force: forceType });
+
+  cy.get(bodySelector).then(($body) => {
+    if (count == 0 || $body.find(checkSelector).is(':visible')) {
+      return;
```

Comment:
shouldn't we fail when no counts are left and the selector is still not visible?

Reply:
Good point!
See my latest commit.

Reply:
ah. took me some time to understand it, but looks good.